google/cloud/bigtable/options.h

```diff
@@ -157,6 +157,42 @@ struct MaxConnectionRefreshOption {
   using Type = std::chrono::milliseconds;
 };
 
+namespace experimental {
+
+/**
+ * If set, the client will throttle mutations in batch write jobs.
+ *
+ * This option is for batch write jobs where the goal is to avoid cluster
+ * overload and prevent job failure more than it is to minimize latency or
+ * maximize throughput.
+ *
+ * With this option set, the server rate-limits traffic to avoid overloading
+ * your Bigtable cluster, while ensuring the cluster is under enough load to
+ * trigger Bigtable [autoscaling] (if enabled).
+ *
+ * The [app profile] associated with these requests must be configured for
+ * [single-cluster routing]. See #google::cloud::bigtable::AppProfileIdOption.
+ *
+ * @note This option must be supplied to `MakeDataConnection()` in order to take
+ * effect.
+ *
+ * @note This feature has only been implemented for the synchronous
+ * `Table::BulkApply()`. It does not have an effect on `Table::AsyncBulkApply()`
+ * or `MutationBatcher::AsyncApply()`.
+ *
+ * @see https://cloud.google.com/bigtable/docs/writes#flow-control
+ *
+ * [autoscaling]: https://cloud.google.com/bigtable/docs/autoscaling
+ * [app profile]: https://cloud.google.com/bigtable/docs/app-profiles
+ * [single-cluster routing]:
+ * https://cloud.google.com/bigtable/docs/routing#single-cluster
+ */
+struct BulkApplyThrottlingOption {
+  using Type = absl::monostate;
```

Comment:
`#include "absl/types/variant.h"`

And did you consider `absl::variant<absl::monostate>`?

Reply:
> `#include "absl/types/variant.h"`

yep

> And did you consider `absl::variant<absl::monostate>`?

Not really. I guess that lets us grow the variant to include a second type. Although if we end up with `absl::variant<absl::monostate, std::chrono::microseconds>`, we might have preferred `absl::optional<std::chrono::microseconds>`...


Hmm.. I changed it.

Reply:
> I guess that lets us grow the variant

Nevermind. We can't grow the variant. This code stops compiling if `Type = absl::variant<absl::monostate, T>`.

```cc
absl::variant<absl::monostate> v = options.get<BulkApplyThrottlingOption>();
```

Reply:
@devbww I reverted the type to `bool`. I will wait for your OK before submitting the PR.

Reply:
I'm not sure I'm claiming things will compile unchanged (although I think they could if someone didn't care about new variants), but rather that the migration path is clear.  But, I can't predict the future (and this wouldn't be high on the list if I could!), so OK.  Thanks for taking a look.